### PR TITLE
Core: add new keywords to include leading zeroes to player number

### PR DESCRIPTION
**Pull request by RadzPrower**
## What is this fixing or adding?
Addition of new keywords (`zplayer`, `ZPLAYER`) which function like their `player` equivalents but with leading zeroes based on the number of total slots generated. This is to aid in searching through a large session of randomly generated slots such as the Big Async. This is done by calculating the number of total digits in the final slot count via `log10 + 1` and then using that value in zfill to ensure that the resulting string is at least `max_digits` long.

The same principle can be applied to the generic `Player{player}` used when using `weights.yaml` should that be desirable but would obviously be hardcoded as it is now rather than relying on the keyword itself.

## How was this tested?
Generated several sessions with 10 or more players using the new keywords to ensure that leading zeroes were included. This includes generating 100 slots with a slightly modified version of Scooty's PR for game-specific names in the AsyncTools which uses the new `zplayer` keyword instead of `player` in the triggers as this was a major impetus for me to make this change. This resulted in up to two leading zeroes and would go up to three in the case of the Big Async:
image

## If this makes graphical changes, please attach screenshots.
N/A

### CODE REVIEW

File: Generate.py [hunk 1]
```diff
@@ -311,6 +316,19 @@ def handle_name(name: str, player: int, name_counter: Counter):
...
+def calculate_digit_length(slot_count: int):
+    # Should never run anything but the initial if statement, but on the off chance 'slot_count'
+    # is a non-positive number for some reason, the elif and else will at least prevent it from
+    # blowing up or throwing an exception.
+    if slot_count > 0:
+        digit_count = int(math.log10(slot_count)) + 1
+    elif slot_count == 0:
+        digit_count = 1
+    else:
+        digit_count = int(math.log10(-slot_count)) + 2
```
Comment (nicholassaylor) [inline Generate.py:328]: Is there a reason why you are using log10 rather than converting to a string and taking the length? It seems like log10 is a bit much for this. 

Additionally, I believe that slot_count being non-positive would cause bigger issues somewhere else and should just throw an exception.

**Review by nicholassaylor — CHANGES_REQUESTED** [review]

File: Generate.py (hunk 1 above)
Comment (RadzPrower) [reply Generate.py:328]: `log10` can be more performant in some scenarios, but you're right that this is a relatively short number and is only performed once, so it should be simplified to that.

As for `slot_count`, in coming back to review the code again after receiving this comment, I noticed that anything less than 1 does nothing anyhow since `player` starts at 1 so the `while` loop started immediately after this new method is called would simply not execute if `args.multi` was anything less than 1. Given this context, a whole method may no longer be necessary as it could likely be replaced by a single line instead as any other values make those edge cases irrelevant.

I will try and pull up my old environment and test those changes before resubmitting.

**Review by EmilyV99 — CHANGES_REQUESTED** [review]
Would having webhost display with leading 0s in the `ID` column solve this issue more simply? I feel like that would be a fine change that would also allow the same level of ctrl+F on the room page, unless there's some part of the issue this is meant to solve I'm not understanding.
